### PR TITLE
[xorg] Adds xcb-util-devel

### PR DESCRIPTION
**Pull request by daravi**
Needed for example for Qt's XCB compilation.

Specify library name and version:  **xorg/system**

- [✓] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [✓] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [✓] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [✓] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
it seems you need to add `libxcb-util-dev` to apt packages (and probably other package managers)

**Comment by Croydon** [comment]
yes, when adding new components they need to be available for all the package manager if possible or the component needs to be somehow conditional

While I can find libxcb packages for Pacman (Arch Linux) and Zypper (Suse), they don't seem to contain `xcb_util.h` 🤔  

https://pkgs.org/search/?q=libxcb

**Comment by Croydon** [comment]
As @ericLemanissier suggested `libxcb-util-dev` seems to be correct for apt 

https://packages.ubuntu.com/focal/amd64/libxcb-util-dev/filelist

File: recipes/xorg/all/conanfile.py [hunk 1]
```diff
@@ -42,22 +42,22 @@ def system_requirements(self):
             if tools.os_info.with_apt:
                 packages = ["xorg-dev", "libx11-xcb-dev", "libxcb-render0-dev", "libxcb-render-util0-dev", "libxcb-xkb-dev",
                             "libxcb-icccm4-dev", "libxcb-image0-dev", "libxcb-keysyms1-dev", "libxcb-randr0-dev", "libxcb-shape0-dev",
-                            "libxcb-sync-dev", "libxcb-xfixes0-dev", "libxcb-xinerama0-dev", "xkb-data"]
+                            "libxcb-sync-dev", "libxcb-xfixes0-dev", "libxcb-xinerama0-dev", "xkb-data", "libxcb-util-dev"]
```
Comment (ericLemanissier) [inline recipes/xorg/all/conanfile.py:45]:
```suggestion
                            "libxcb-sync-dev", "libxcb-xfixes0-dev", "libxcb-xinerama0-dev", "xkb-data", "libxcb0-util-dev" if tools.os_info.linux_distro == "ubuntu" and tools.os_info.os_version <= "14" else "libxcb-util-dev"]
```
gcc 4.9 docker image is based on trusty, which does not have `libxcb-util-dev` but `libxcb0-util-dev` while more recent distributions don't have `libxcb0-util-dev`

**Comment by daravi** [comment]
Does `tools.os_info.os_version` return something like 14.x.y? Or just 14?

File: recipes/xorg/all/conanfile.py [hunk 2]
```diff
@@ -42,22 +42,22 @@ def system_requirements(self):
             if tools.os_info.with_apt:
                 packages = ["xorg-dev", "libx11-xcb-dev", "libxcb-render0-dev", "libxcb-render-util0-dev", "libxcb-xkb-dev",
                             "libxcb-icccm4-dev", "libxcb-image0-dev", "libxcb-keysyms1-dev", "libxcb-randr0-dev", "libxcb-shape0-dev",
-                            "libxcb-sync-dev", "libxcb-xfixes0-dev", "libxcb-xinerama0-dev", "xkb-data"]
+                            "libxcb-sync-dev", "libxcb-xfixes0-dev", "libxcb-xinerama0-dev", "xkb-data", "libxcb0-util-dev" if tools.os_info.linux_distro == "ubuntu" and tools.os_info.os_version <= "14" else "libxcb-util-dev"]
```
Comment (ericLemanissier) [inline recipes/xorg/all/conanfile.py:45]:
```suggestion
                            "libxcb-sync-dev", "libxcb-xfixes0-dev", "libxcb-xinerama0-dev", "xkb-data", "libxcb0-util-dev" if tools.os_info.linux_distro == "ubuntu" and tools.os_info.os_version < "15" else "libxcb-util-dev"]
```
sorry, trusty is actually 14.04, which is not <= 14 ! This time, I tested it :)

Comment (ericLemanissier) [reply recipes/xorg/all/conanfile.py:45]: well, don't commit this yet : `dpkg-query: no packages found matching libxcb0-util-dev`

Comment (daravi) [reply recipes/xorg/all/conanfile.py:45]: Thanks! By the way would this work on 13 and lower or is that too old for us to care?

Comment (ericLemanissier) [reply recipes/xorg/all/conanfile.py]: I have no idea, but this distribution is not part of CCI test configurations, so there are most probably a LOT of other problems on all recipes.

**Review by ericLemanissier — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by Croydon — APPROVED** [review]

**Pull request merged**